shub/deploy.py

```diff
@@ -75,7 +75,7 @@
 @click.option("-k", "--keep-log", help="keep the deploy log", is_flag=True)
 def cli(target, version, debug, egg, build_egg, verbose, keep_log):
     conf, image = load_shub_config(), None
-    if target in conf.projects:
+    if target != 'default':
```

Comment:
not sure I follow why we have this check at all. shouldn't we try to get target config always, use image if configured, catch error and set `image = None` if not cannot find target?

```
(Pdb) conf.get_target_conf(108106)
Target(project_id=108106, endpoint='https://app.scrapinghub.com/api/', apikey='f872157bf3f2481a9c58b7dc799ff836', stack=None, image='images.scrapinghub.com/project/108106', requirements_file=None, version='f092796-fix-docker-disconnects', eggs=[])
(Pdb) conf.get_target_conf(108107)
Target(project_id=108107, endpoint='https://app.scrapinghub.com/api/', apikey='f872157bf3f2481a9c58b7dc799ff836', stack=None, image='images.scrapinghub.com/project/108107', requirements_file=None, version='f092796-fix-docker-disconnects', eggs=[])
(Pdb) conf.get_target_conf('asdf')
*** shub.exceptions.BadParameterException: Could not find target "asdf". Please define it in your scrapinghub.yml or supply a numerical project ID.
```